lib/thin/sqlnet/ntTcp.js

```diff
@@ -178,7 +178,7 @@ class NTTCP {
         });
         req.end();
       } else {
-        this.stream = net.connect(address.port, address.host, () => {
+        this.stream = net.connect(Number(address.port), String(address.host), () => {
```

Comment:
It would be better if the source of `address.port` and `address.host` were adjusted as needed.